v3/src/hooks/use-overlay-bounds.ts

```diff
@@ -1,36 +1,19 @@
-import { useEffect, useState } from "react"
-
 interface IProps {
   target: Element | null
   portal?: Element | null
 }
-interface IBounds {
-  left: number
-  top: number
-  width: number
-  height: number
-}
-export function useOverlayBounds({ target, portal }: IProps) {
-  const [overlayBounds, setOverlayBounds] = useState<IBounds | null>(null)
 
-  useEffect(() => {
-    // track the bounds of the target element relative to the portal element
-    const observer = target && new ResizeObserver(() => {
-      const portalBounds = portal?.getBoundingClientRect()
-      const targetBounds = target?.getBoundingClientRect()
-      if (targetBounds) {
-        setOverlayBounds({
-          left: targetBounds.x - (portalBounds?.x ?? 0),
-          top: targetBounds.y - (portalBounds?.y ?? 0),
-          width: targetBounds.width,
-          height: targetBounds.height
-        })
-      }
-    })
-    target && observer?.observe(target)
-
-    return () => observer?.disconnect()
-  }, [portal, target])
-
-  return overlayBounds
+export function useOverlayBounds({ target, portal }: IProps) {
+  const portalBounds = portal?.getBoundingClientRect()
+  const targetBounds = target?.getBoundingClientRect()
+  if (targetBounds) {
+    return {
+      left: targetBounds.x - (portalBounds?.x ?? 0),
+      top: targetBounds.y - (portalBounds?.y ?? 0),
+      width: targetBounds.width,
+      height: targetBounds.height
+    }
+  } else {
+    return {}
```

Comment:
@kswenson Should I be returning null? Compiler at calls to this hook will be mad if I don't return something, so this seemed the most compatible, but still doesn't look quite right?

Reply:
@kswenson it turns out that since we were returning the value into a variable declared as (React's) type `CSSProperties`, it will not accept `null`, but will accept `{}`.  So I changed it back.  

If you think there is a reason to not type the style variable in `droppable-svg` like this: 

```ts const style: CSSProperties = useOverlayBounds({ target, portal })```

or make some other change, let me know.

Reply:
That's fine -- when I made that suggestion I was thinking it was going to be rendered directly. `{}` is the right default for `CSSProperties`.